fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/MergeTwoAggRule.java

```diff
@@ -37,7 +37,7 @@
 public class MergeTwoAggRule extends TransformationRule {
     private static final List<String> ALLOW_MERGE_AGGREGATE_FUNCTIONS = Lists.newArrayList(FunctionSet.SUM,
             FunctionSet.MAX, FunctionSet.MIN, FunctionSet.BITMAP_UNION, FunctionSet.HLL_UNION,
-            FunctionSet.PERCENTILE_UNION, FunctionSet.ANY_VALUE);
+            FunctionSet.PERCENTILE_UNION, FunctionSet.ANY_VALUE, FunctionSet.MAX_BY);
```

Comment:
I think MAX_BY function doesn't support this rule.

Reply:
> I think MAX_BY function doesn't support this rule.

fixed